src/module.c

```diff
@@ -7709,16 +7962,14 @@ void moduleHandleBlockedClients(void) {
             reply_us = elapsedUs(replyTimer);
             moduleFreeContext(&ctx);
         }
-
+        /* Hold onto the blocked client if custom auth is in progress. The reply callback is invoked
+         * when the client is reprocessed. */
+        if (c && clientHasModuleAuthInProgress(c)) {
+            c->module_blocked_client = bc;
+        }
         /* Free privdata if any. */
-        if (bc->privdata && bc->free_privdata) {
-            RedisModuleCtx ctx;
-            int ctx_flags = c == NULL ? REDISMODULE_CTX_BLOCKED_DISCONNECTED : REDISMODULE_CTX_NONE;
-            moduleCreateContext(&ctx, bc->module, ctx_flags);
-            ctx.blocked_privdata = bc->privdata;
-            ctx.client = bc->client;
-            bc->free_privdata(&ctx,bc->privdata);
-            moduleFreeContext(&ctx);
+        else {
```

Comment:
styling nit pick, no newline between `{` and `else`.
i guess in this case the comment should go into the `else` block.

Reply:
Addressed here: https://github.com/redis/redis/pull/11659/commits/457f36047e7817312bac9fc09a8d8ad74547a3f2